### PR TITLE
fix(graphene): crash in allForms query

### PR DESCRIPTION
**Pull request by JohnL17**

### CODE REVIEW

File: caluma/caluma_form/tests/test_form.py [hunk 1]
```diff
@@ -63,6 +63,52 @@ def test_query_all_forms(
...
+                }
+              }
+            }
+          }
+        }
+    """
+
+    result = schema_executor(
+        query,
+    )
```
Comment (winged) [inline caluma/caluma_form/tests/test_form.py:106]: You're not passing the `$slug` input variable here - is this intentional?

This causes the `slug` filter to be disregarded, causing the error.

Comment (winged) [reply caluma/caluma_form/tests/test_form.py:106]: Added a check in the filter code to avoid crash

File: caluma/caluma_form/tests/test_form.py [hunk 2]
```diff
@@ -63,6 +63,52 @@ def test_query_all_forms(
...
+                name
+                meta
+                isPublished
+                source {
+                  slug
+                }
+              }
+            }
+          }
+        }
```
Comment (anehx) [inline caluma/caluma_form/tests/test_form.py:101]: Let's keep the test simple:

```suggestion
        query Form($slug: String) {
          allForms(filter: [{ slug: $slug }]) {
            totalCount
          }
        }
```

File: caluma/caluma_form/tests/test_form.py [hunk 3]
```diff
@@ -63,6 +63,52 @@ def test_query_all_forms(
...
+    form,
+    form_factory,
+    form_question,
+    form_question_factory,
+    question,
+    schema_executor,
+):
+    form_factory(name="3rd", description="Second result")
+    form_factory(name="2nd", description="Second result")
+    form_question_factory(form=form)
```
Comment (anehx) [inline caluma/caluma_form/tests/test_form.py:78]:
```suggestion
    form_factory.create_batch(5)
```

File: caluma/caluma_form/tests/test_form.py [hunk 4]
```diff
@@ -63,6 +63,52 @@ def test_query_all_forms(
...
+def test_query_root_forms(
+    db,
+    snapshot,
+    form,
+    form_factory,
+    form_question,
+    form_question_factory,
+    question,
+    schema_executor,
+):
```
Comment (anehx) [inline caluma/caluma_form/tests/test_form.py:75]:
```suggestion
def test_optional_filter(db, form_factory, question, schema_executor):
```

File: caluma/caluma_form/tests/test_form.py [hunk 5]
```diff
@@ -63,6 +63,52 @@ def test_query_all_forms(
...
+          }
+        }
+    """
+
+    result = schema_executor(
+        query,
+    )
+
+    assert not result.errors
+    snapshot.assert_match(result.data)
```
Comment (anehx) [inline caluma/caluma_form/tests/test_form.py:109]:
```suggestion
    assert result.data["allForms"]["totalCount"] == 5
```

**Review by anehx — CHANGES_REQUESTED** [review]
Let's keep the test generic and simple..

**Comment by winged** [comment]
@anehx Thanks for the review, applied your suggestions

**Review by anehx — APPROVED** [review]

**Pull request merged**